mark-larter/ad-nimbus
# Nginx proto

The base now runs basic JMeter tests with some manual setup.

Changes
1. nginx and net-location use Docker volumes rather than loading the files into the container. This eliminates having to remember to rebuild the containers whenever you tweak index.js or nginx.conf
2. nginx and net-location work together. 
3. coreos IP addresses now match the instance number. 172.17.8.104 is the IP address for coreos-04.
4. Various functions and aliases added to make it easier to build&push containers and start services. Note: the docker hub has asteere hardcoded (future improvement).
5. The maxmind files are now included in git, previously they were in the container. 
6. In case I missed it, can you check that there isn't anything proprietary (data files) in the pull request.
7. Add JMeter test plan

How to run after the merge
1. Recreate the coreos
vdestroy
v up
vsh1
. share/.profile
1. Start the services
   startnet-location
   startnginx
   fluf - list the services running in the cluster. 4 net-location and 1 nginx service should be launched
   flm - to list the machines and their IP addresses
2. Start checking the status of the services until they are running (pulling the containers takes awhile). 
3. To get the status of the services on the other coreos 
   Open a new terminal window and vsh\* to the coreos you care about
   . share/.profile
   f status <net-location|nginx>@<instance>.service
4. Once the nginx and one of the net-location services has started, open your favorite browser to the IPaddressOfCoreOSRunningNginxService:49160. The hex value of the hostname should change each time you reload the page cycling through the net-location containers.
